### PR TITLE
Parser: recover on unfinished nested modules

### PR DESCRIPTION
**Pull request by auduchinok**
Various unfinished nested module cases:

```fsharp
namespace Ns

module
```

```fsharp
namespace Ns

module rec
```

```fsharp
namespace Ns

module A
```

```fsharp
namespace Ns

module A =
```

Recovery for subsequent declarations is possible due to the new rules:
```fsharp
namespace Ns

module A =

module B =
    ()
```

### CODE REVIEW

File: src/Compiler/pars.fsy [hunk 1]
```diff
@@ -521,6 +521,12 @@ moduleIntro:
         let mModule = rhs parseState 1
         mModule, $4, $5.LongIdent, $3, $2 }
 
+  | moduleKeyword opt_attributes opt_access opt_rec OBLOCKSEP
+      { if not (isNil $2) then
+            parseState.LexBuffer.CheckLanguageFeatureAndRecover LanguageFeature.AttributesToRightOfModuleKeyword (rhs parseState 4)
```
Comment (T-Gro) [inline src/Compiler/pars.fsy:526]: Do we have a test for this logical condition around LanguageFeature.AttributesToRightOfModuleKeyword somewhere?

File: src/Compiler/SyntaxTree/LexFilter.fs [hunk 2]
```diff
@@ -1969,7 +1972,8 @@ type LexFilterImpl (
         | MODULE, _ :: _ ->
             insertComingSoonTokens("MODULE", MODULE_COMING_SOON, MODULE_IS_HERE)
             if debug then dprintf "MODULE: entering CtxtModuleHead, awaiting EQUALS to go to CtxtSeqBlock (%a)\n" outputPos tokenStartPos
-            pushCtxt tokenTup (CtxtModuleHead (tokenStartPos, token, NotLexingModuleAttributes))
+            let isNested = match offsideStack with | [ CtxtSeqBlock _ ] -> false | _ -> true
```
Comment (T-Gro) [inline src/Compiler/SyntaxTree/LexFilter.fs:1975]: With that, is

```namespace N

module M =```

also nested?

File: src/Compiler/pars.fsy (hunk 1 above)
Comment (auduchinok) [reply src/Compiler/pars.fsy:526]: Yes, it looks like that some tests were added with the original rule in https://github.com/dotnet/fsharp/pull/11722.

File: src/Compiler/SyntaxTree/LexFilter.fs (hunk 2 above)
Comment (auduchinok) [reply src/Compiler/SyntaxTree/LexFilter.fs:1975]: Correct, that's the terminology from the `SyntaxTree`:
https://github.com/dotnet/fsharp/blob/e267bb9f8d590feed1b94b469d78cfce61afecad/src/Compiler/SyntaxTree/SyntaxTree.fs#L1461

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**